pkg/cluster/mpi/configmap.go

```diff
@@ -14,12 +15,47 @@ import (
 	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/core"
 )
 
-var (
-	sshdConfigTmpl = `
-Port %d
-AuthorizedKeysFile %s
-`
-)
+const launchScriptTemplate = `#!/bin/bash
+
+set -o nounset 
+set -o errexit
+
+SSH_PORT=%[1]d
+DOMINO_UID=%[2]d
+DOMINO_GID=%[3]d
+DOMINO_USER=%[4]s
+DOMINO_GROUP=%[5]s
+AUTHORIZED_KEYS_PATH=%[6]s
+
+if ! id $DOMINO_UID >/dev/null 2>&1; then
+    groupadd -g $DOMINO_GID $DOMINO_GROUP
+    useradd -u $DOMINO_UID -g $DOMINO_GID -mN -s /bin/bash $DOMINO_USER
+fi
+if [ "$(id -nu $DOMINO_UID)" != "$DOMINO_USER" ]; then
+    echo >&2 "User name mismatch"
+    exit 1
+fi
+
+DOMINO_HOME=$(eval echo "~$DOMINO_USER")
+SSH_DIR="$DOMINO_HOME/.ssh"
+mkdir -p "$SSH_DIR"
+
+ssh-keygen -f "$SSH_DIR/ssh_host_key" -N '' -t rsa
```

Comment:
https://blog.trailofbits.com/2019/07/08/fuck-rsa/

I think we should use ecdsa - the same standard we use to generate the user key pair

Reply:
Just checked that `ssh-keygen` supports ecdsa, so fine with me.